internal/app/go-filecoin/internal/submodule/storage_mining_submodule.go

```diff
@@ -78,7 +78,7 @@ func NewStorageMiningSubmodule(minerAddr address.Address, ds datastore.Batching,
 		PoStGenerator:    postGen,
 		minerNode:        minerNode,
 		storageMiner:     storageMiner,
-		heaviestTipSetCh: c.HeaviestTipSetCh,
+		heaviestTipSetCh: make(chan interface{}),
```

Comment:
Nothing was ever coming out of this channel because its already being consumed. Create a new channel here and feed it in HandleNewHead method.

Reply:
ugh, we have to get rid of these channels.